packages/patterns/README.md

```diff
@@ -1,103 +1,32 @@
-`NaN` means "incomparable" --- the first key is not less than, equivalent to, or greater than the second.
-To keep the orders distinct when speaking informally, we use "earlier" and "later" for rank order, and "smaller" and "bigger" for key order.
-
-The key ordering of `compareKeys` refines the rank ordering of `compareRank` but leaves gaps for which a more complete "full order" relies upon rank ordering:
-1. `compareKeys(X,Y) === 0` implies that `compareRank(X,Y) === 0` --- if X
-   is equivalent to Y in key order, then X is equivalent to Y in rank order.
-   But the converse does not hold; for example, Remotables `Far('X')` and
-   `Far('Y')` are equivalent in rank order but incomparable in key order.
-2. `compareKeys(X,Y) < 0` implies that `compareRank(X,Y) < 0` --- if X is
-   smaller than Y in key order, then X is earlier than Y in rank order.
-   But the converse does not hold; for example, the record `{b: 3, a: 5}`
-   is earlier than the record `{b: 5, a: 3}` in rank order but they are
-   incomparable in key order.
-3. `compareRank(X,Y) === 0` implies that `compareKeys(X,Y)` is either
-   0 or NaN --- Keys within the same rank are either equivalent to or
-   incomparable to each other in key order. But the converse does not hold;
-   for example, `Far('X')` and `{}` are incomparable in key order but not
-   equivalent in rank order.
-4. `compareRank(X,Y) === 0` and `compareRank(X,Z) === 0` imply that
-   `compareKeys(X,Y)` and `compareKeys(X,Z)` are the same --- all Keys within
-   the same rank are either mutually equivalent or mutually incomparable, and
-   in fact only in the mutually incomparable case can the rank be said to
-   contain more than one key.
-
-## Relationships between types
+mustMatch(specimen, pattern); // throws: 'bar?: number 4 - Must be a string'
+```
 
-The set of all primitive values is a strict subset of Data, which is a strict subset of Keys.
+`M` also has {@link GuardMakers} methods to make {@link InterfaceGuard}s that use Patterns to characterize dynamic behavior such as method argument/response signatures and promise awaiting. The {@link @endo/exo!} package uses `InterfaceGuard`s (each of which maps a collection of method names to their respective method guards) as the first level of defense for Exo objects against malformed input.
 
-The set of all primitive values is also a strict subset of the set of Scalars (which is the union of Primitives and Capabilities [i.e., Remotables and Promises]). The union of primitive values and Remotables is a strict subset of Keys.
 
-Keys is a strict subset of Patterns, which is a strict subset of Passables.
+## Key Equality, Containers
 
-TODO: Include a diagram visually demonstrating the following.
+Builds on {@link @endo/pass-style!} as described in [`kindOf` and `passStyleOf` levels of abstraction](./docs/marshal-vs-patterns-level.md) to define higher level data types as individual refinements of Passable CopyTagged records (PassStyle "tagged"):
```

Comment:
I don't suppose this sort of link markup works in the https://www.npmjs.com/package/@endo/patterns context.

We should have a link from there to https://endojs.github.io/endo/modules/_endo_patterns.html or at least to https://endojs.github.io/, I suppose.

Reply:
`@link` is JSDoc: https://jsdoc.app/tags-inline-link